web/src/copy_messages.ts

```diff
@@ -263,12 +247,12 @@ export function copy_handler(): boolean {
 
     // Select div so that the browser will copy it
     // instead of copying the original selection
-    insert_and_select_div($div, selection);
-    // eslint-disable-next-line @typescript-eslint/no-deprecated
-    document.execCommand("copy");
-    setTimeout(() => {
-        restore_original_selection(ranges);
 
+    const html_content = $div.html().trim();
+    const plain_text = $div.text().trim();
+    ev.clipboardData?.setData("text/html", html_content);
+    ev.clipboardData?.setData("text/plain", plain_text);
+    setTimeout(() => {
         $div.remove();
```

Comment:
Is this still necessary? I'm not sure `$div` will have been added to the DOM at all, given `insert_and_select_div` is no longer called.